### PR TITLE
change read from file to read from url

### PR DESCRIPTION
**Pull request by thphuongster**
fetch data from url
Screenshot 2023-02-19 190408

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
+        .then((response) => response.json())
+        .then((data: Item[]) => {
+          setTasks(
+            data.map((item) => ({
+              id: item.id,
+              name: item.name,
+              isDeleted: false,
+            }))
+          );
+          console.log("Task ...", tasks);
```
Comment (ptmdmusique) [inline src/App.tsx]: we can delete this

File: src/App.tsx [hunk 2]
```diff
@@ -1,21 +1,44 @@
...
+  //    Fetch data
+    const [tasks, setTasks] = useState<ExItem[]>([]);
+    
+    useEffect(() => {
+      fetch("https://www.jsonkeeper.com/b/MF8H")
+        .then((response) => {
+          if (!response.ok) {
+            throw new Error(response.statusText);
+          }
+          return response;
```
Comment (ptmdmusique) [inline src/App.tsx]: we can also do `return response.json()` here instead of adding another `.then` below

File: src/App.tsx [hunk 3]
```diff
@@ -1,21 +1,44 @@
...
+
+  interface Item {
+    id: number;
+    name: string;
+  }
+  interface ExItem {
+    id: number;
+    name: string;
+    isDeleted: boolean;
+  }
```
Comment (ptmdmusique) [inline src/App.tsx]: let's bring these outside of the component

File: src/App.tsx [hunk 4]
```diff
@@ -1,21 +1,44 @@
...
-      isDeleted: false,
-    }))
-  );
-  const [newTaskName, setNewTaskName] = useState("");
+
+  interface Item {
+    id: number;
+    name: string;
+  }
+  interface ExItem {
```
Comment (ptmdmusique) [inline src/App.tsx]: this name is not clear, what does `Ex` mean? Maybe `ItemWithDeleteFlag` is a better name

**Review by ptmdmusique — APPROVED** [review]
nice work! you can merge it

**Comment by thphuongster** [comment]
useeffect cleanup return changed

**Pull request merged**